### PR TITLE
MM-61198: react-hooks-testing-library and renderHookWithContext

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary

Adds https://github.com/testing-library/react-hooks-testing-library and `renderHookWithContext()` for use in testing custom hooks.

`@testing-library/react-hooks` is only needed for `react@17` and `testing-library@12` and down; it is redundant with `react@18` and `testing-library@13` and above.

#### Ticket Link
 - https://mattermost.atlassian.net/browse/MM-61198

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]
SHIP IT :smile:

File: webapp/channels/src/components/threading/hooks.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            },
+            teams: {
+                currentTeamId: mockTeam.id,
+            },
+        },
+    };
+
+    describe('useThreadRouting', () => {
+        test('should indicate current team and user', () => {
+            const {result} = renderHookWithContext(() => useThreadRouting(), mockState);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/threading/hooks.test.ts:30]: i am assuming the return types wont match for the actual hooks? or it does match the return type of the hook

File: webapp/package-lock.json [hunk 2]
```diff
@@ -227,6 +228,121 @@
         "yargs": "16.2.0"
       }
     },
+    "channels/node_modules/@mattermost/mmjstool": {
```
Comment (M-ZubairAhmed) [inline webapp/package-lock.json:231]: i think this wasnt intentional change, but its fine i think we didnt update the lock file for a while

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/threading/hooks.test.ts (hunk 1 above)
Comment (calebroseland) [reply webapp/channels/src/components/threading/hooks.test.ts:30]: The type of `result.current` should match the return type of the hook

File: webapp/package-lock.json (hunk 2 above)
Comment (calebroseland) [reply webapp/package-lock.json]: nope, reran it and not there. I think I ran an odd i18n-extract command in a different directory or something.

**Review by crspeller — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Awesome! I think I said it last week, but always assumed this was blocked on React 18. I never thought to check for `renderHook` in a standalone package

**Pull request merged**